polaris/dataset/zarr/_checksum.py

```diff
@@ -0,0 +1,410 @@
+"""
```

Comment:
Is the intention to keep this up to date with the `zarr_checksum` lib as it evolves and receives updates? Or do we really want to fork it and freeze it at this point in time? If we want to keep up with any updates in the original source lib this will make it very difficult (or impossible). But maybe that doesn't matter. If it does, though, another option could be to fork the `zarr_checksum` library and use our fork as dependency, then try to organize changes we make in a way that will minimize merge conflicts with any upstream changes, so we can pull them in as desired.

Reply:
We have already deviated from the original implementation and if anything, I expect us to deviate more in the future. It also doesn't seem like the maintainers of the upstream package are working towards an exciting set of new features we would want to pull in. The last commit was from 6 months ago. 

I thus don't think we would want to receive updates from the upstream package. I could see this be its own package at some point, but for now I consider it small enough that I would prefer to keep it all in a single repo to simplify logistics (e.g. no need to coordinate PRs between repositories).